open_humans/management/commands/vacuum_log_bucket.py

```diff
@@ -0,0 +1,112 @@
+                fields = [
+                    "bucket_owner",
+                    "bucket",
+                    "time",
+                    "remote_ip",
+                    "requester",
+                    "request_id",
+                    "operation",
+                    "bucket_key",
+                    "request_uri",
+                    "status",
+                    "error_code",
+                    "bytes_sent",
+                    "object_size",
+                    "total_time",
+                    "turn_around_time",
+                    "referrer",
+                    "user_agent",
+                    "version_id",
+                    "host_id",
+                    "signature_version",
+                    "cipher_suite",
+                    "auth_type",
+                    "host_header",
+                ]
+                for index, field_name in enumerate(fields):
+                    field = aws_log_entry._meta.get_field(field_name)
+                    if field.get_internal_type() == "IntegerField":
+                        log_item = log[index]
+                        if (log_item == "-") or (log_item == '"-"'):
+                            log_item = 0
+                        log[index] = int(log_item)
+                    if field.get_internal_type() == "DateTimeField":
+                        log[index] = datetime.strptime(
+                            log[index], "[%d/%b/%Y:%H:%M:%S %z]"
+                        )
+                    if index == 17:
+                        # Sometimes, aws inserts a stray '-' here, klugey workaround
+                        if (log[17] == "-") and (len(log[18]) < 32):
+                            # The actual Host ID is always quite long
+                            log.pop(17)
+                    setattr(aws_log_entry, field_name, log[index])
+
+                data_file = DataFile.objects.filter(file=aws_log_entry.bucket_key)
+                if data_file:
+                    aws_log_entry.data_file = data_file.get()
+                url = aws_log_entry.request_uri.split(" ")[1]
+
+                # Filter out things we don't care to log
+                if settings.AWS_STORAGE_BUCKET_NAME in url:
+                    continue
+                if any(
+                    blacklist_item in url for blacklist_item in AWS_LOG_KEY_BLACKLIST
+                ):
+                    continue
+                aws_log_entry.save()
+
+                # Associate with any potential access logs from the Open Humans end.
+
+                # Split out the key from the url
+                parsed_url = urlparse(url)
+                # parse_qs returns a dict with lists as values
+                oh_key = parse_qs(parsed_url.query).get("x-oh-key", [""])[0]
+                if oh_key == "None":
+                    oh_key = None
+                if oh_key:
+                    oh_data_file_access_logs = NewDataFileAccessLog.objects.filter(
+                        key=oh_key
```

Comment:
per slack conversation, it seems like this was the reason to shift away from JSONFIELD... maybe try sticking with JSONField, I think filtering is still possible?
https://stackoverflow.com/questions/36389871/django-jsonfield-filtering